mono/mini/mini-trampolines.c

```diff
@@ -824,19 +824,19 @@ gpointer
 mono_magic_trampoline (mgreg_t *regs, guint8 *code, gpointer arg, guint8* tramp)
 {
 	gpointer res;
+	MonoError error;
 
-	MONO_ENTER_GC_UNSAFE_UNBALANCED;
+	MONO_REQ_GC_UNSAFE_MODE;
 
-	MonoError error;
+	g_assert (mono_thread_is_gc_unsafe_mode ());
```

Comment:
This is redundant with https://github.com/mono/mono/pull/5409/files#diff-9d04cbdf08cdb766439413f26549f9c7R829

Reply:
What it is redundant with ? The MONO_REQ_GC_UNSAFE_MODE line is only executed in checked mode.